modules/programs/kodi.nix

```
@@ -0,0 +1,238 @@
+  '';
+
+  stylesheetSourcesRootTag = ''
+    <xsl:template match='/'>
+      <xsl:comment> Generated by Home Manager. </xsl:comment>
+      <sources>
+        <xsl:apply-templates/>
+      </sources>
+    </xsl:template>
+  '';
+
+  stylesheetAddonSettingsRootTag = ''
+    <xsl:template match='/'>
+      <xsl:comment> Generated by Home Manager. </xsl:comment>
+      <settings version="2">
+        <xsl:apply-templates/>
+      </settings>
+    </xsl:template>
+  '';
+
+  attrsetToXml = attrs: name: stylesheet:
+    pkgs.stdenv.mkDerivation {
+      inherit name stylesheet;
+      nativeBuildInputs = [ pkgs.libxslt ];
+      phases = [ "installPhase" ];
+      installPhase = let xml = (builtins.toXML attrs);
+      in ''
+        xsltproc ${stylesheet} - <<EOF > $out
+        ${xml}
+        EOF
+      '';
+    };
+
+  attrsetToAdvancedSettingsXml = attrs: name:
+    let
+      stylesheet = builtins.toFile "stylesheet.xsl" ''
+        ${stylesheetCommonHeader}
+        ${stylesheetAdvancedSettingsRootTag}
+        ${stylesheetNestedTags}
+        ${stylesheetCommonFooter}
+      '';
+    in (attrsetToXml attrs name stylesheet);
+
+  attrsetToSourcesXml = attrs: name:
+    let
+      stylesheet = builtins.toFile "stylesheet.xsl" ''
+        ${stylesheetCommonHeader}
+        ${stylesheetSourcesRootTag}
+        ${stylesheetNestedTags}
+        ${stylesheetCommonFooter}
+      '';
+    in (attrsetToXml attrs name stylesheet);
+
+  attrsetToAddonSettingsXml = attrs: name:
+    let
+      stylesheet = builtins.toFile "stylesheet.xsl" ''
+        ${stylesheetCommonHeader}
+        ${stylesheetAddonSettingsRootTag}
+        ${stylesheetTagsAsSettingWithId}
+        ${stylesheetCommonFooter}
+      '';
+    in (attrsetToXml attrs name stylesheet);
+
+in {
+  meta.maintainers = [ maintainers.dwagenk ];
+
+  options.programs.kodi = {
+    enable = mkEnableOption "Kodi";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.kodi;
+      defaultText = literalExample "pkgs.kodi";
+      example = literalExample ''
+        { pkgs.kodi.withPackages (exts: [ exts.pvr-iptvsimple ]) }
+      '';
+      description = ''
+        The <literal>kodi</literal> package to use.
+        Can be used to specify extensions.
+      '';
+    };
+
+    datadir = mkOption {
+      type = types.path;
+      default = "${config.home.homeDirectory}/.kodi";
+      defaultText = literalExpression "\"\${config.home.homeDirectory}/.kodi\"";
+      description = "Directory to store configuration and metadata.";
+      example = literalExpression "\"\${config.xdg.dataHome}/kodi\"";
+    };
+
+    settings = mkOption {
+      type = with types; nullOr (attrsOf anything);
+      default = null;
+      description = ''
+        Configuration to write to the <literal>advancedsettings.xml</literal>
+        file in kodis userdata directory. Settings specified here will be
+        immutable from inside kodi and be hidden from the GUI settings dialog.
+
```

Need
```suggestion
        </para><para>
```
to indicate a paragraph.